### PR TITLE
Feature/number field sub types

### PR DESCRIPTION
**Pull request by ConnorStroomberg**
Add integer, long and decimal subtypes to improve input validation

\+ Add separate demo pages for field types to improve maintainability and testability  

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- Clean commits
- [x] No warnings during install
- [x] Updated flow typing

### CODE REVIEW

File: src/util/EntityToFormMapper.js [hunk 1]
```diff
@@ -248,17 +276,22 @@ const generateFormSchemaField = (attribute, mapperOptions?: MapperOptions): Form
   // options is a function that always returns an array of option objects
   const options = getFieldOptions(attribute, mapperOptions)
   let fieldProperties = {
-    type: getHtmlFieldType(attribute.fieldType),
     id: attribute.name,
     label: attribute.label,
     description: attribute.description,
+    type: getHtmlFieldType(attribute.fieldType),
     required: isRequired(attribute),
     disabled: attribute.readOnly || attribute.fieldType === 'ONE_TO_MANY',
```
Comment (marikaris) [inline src/util/EntityToFormMapper.js]: in the "old" forms the one to many is not disabled and in some magic way, it is also functional

File: src/util/EntityToFormMapper.js [hunk 2]
```diff
@@ -248,17 +276,22 @@ const generateFormSchemaField = (attribute, mapperOptions?: MapperOptions): Form
...
     readOnly: attribute.readOnly || attribute.fieldType === 'ONE_TO_MANY',
```
Comment (marikaris) [inline src/util/EntityToFormMapper.js]: in the "old" forms the one to many is not readonly and in some magic way, it is also functional

File: src/components/field-types/TypedFieldComponent.vue [hunk 3]
```diff
@@ -73,20 +77,42 @@
...
         }
-        return customValidation
+
+        return validate
+      },
+      stepSize () {
+        // Conditionally add step size, return false to omit step attribute
+        const fieldType = this.field.type
+        const fieldSubType = this.field.subType
+        return fieldType === 'number' && (fieldSubType === 'integer' || fieldSubType === 'long') ? 1 : false
```
[the remark this answers is not in the document]
Comment (ConnorStroomberg) [reply src/components/field-types/TypedFieldComponent.vue]: Step size is set to 1 for integer ( 1, 2, 3....) or to false to indicate no step size attribute should be set. ( see comment in code)

File: src/util/EntityToFormMapper.js (hunk 1 above)
Comment (ConnorStroomberg) [reply src/util/EntityToFormMapper.js]: Have to look in to this, i do not know why this is , i did not change it in the PR, suggest to leave it as is for now.

**Pull request merged**